BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt

```diff
@@ -69,111 +74,26 @@ open class BraintreeClient @VisibleForTesting internal constructor(
      */
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     constructor(clientParams: ClientParams) :
```

Comment:
> The effect of this change for merchants is that if they pass an invalid authentication string to their client constructors, it will be returned as a config error (error fetching configuration, rather than error fetching authorization).

On iOS, the `BTAPIClient` (closest to android's `BraintreeClient`) has a [failable init](https://github.com/braintree/braintree_ios/blob/d422ab673680fc8cbf2c6a333f3dc2ebb79fb426/Sources/BraintreeCore/BTAPIClient.swift#L65) which gets triggered if some of the SDKs manual checks on the clientToken after base64 decoding fail.

We _could_ move up that logic in the Android sequence of events to more closely mirror iOS and return an error sooner if the auth is invalid (I think it lives [here](https://github.com/braintree/braintree_android/blob/ed09dabd22f8f594088ce5659a800e7c4309abde/BraintreeCore/src/main/java/com/braintreepayments/api/Authorization.kt#L48-L54)). Total take it or leave it comment - curious your thoughts

Reply:
The concept of a failable init doesn't really exist in Kotlin or Java - there are some hacky ways to achieve that behavior but I don't think it would be an expected integration pattern for merchants. We could always throw an error in the `init` method, but that means merchants have to handle the error which I think is less ideal than calling back an error further in the flow. To make it more explicit, we could add a check to the four BraintreeClient methods  that fail for auth (http calls and getConfiguration) if auth is `InvalidAuthorization` and callback an authorization error at the top of those methods so that the errors are more clear. Requires some code duplication but I think that's probably the best merchant experience - what do you think?

Reply:
@scannillo I decided to go ahead and callback an auth error in the four BraintreeClient methods that would error on config fetch for invalid auth. In v4 we had a helpful auth error message directing merchants to docs so I think that's a good message to keep.